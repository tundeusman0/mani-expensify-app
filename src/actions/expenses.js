import uuid from 'uuid'
// ADD_EXPENSES
export const add_expenses = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})
// REMOVE_EXPENSES
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
})
// EDIT_EXPENSES
export const edit_expenses = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
})