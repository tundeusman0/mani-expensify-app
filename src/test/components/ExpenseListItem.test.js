import React from 'react'
import {shallow} from 'enzyme';
import ExpenselistItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses';

test('should render ExpenseListItem with expenses',()=>{
    const wrapper = shallow(<ExpenselistItem expenses={{...expenses[0]}}/>)
    expect(wrapper).toMatchSnapshot()
})
