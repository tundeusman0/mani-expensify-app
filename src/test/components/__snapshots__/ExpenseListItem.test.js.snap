// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <Link
    replace={false}
    to="/editPage/undefined"
  >
    <h3 />
  </Link>
  <p>
    0.00
    -
    Jan 1st,1970
  </p>
</div>
`;
